### PR TITLE
Add react class element order rules

### PR DESCRIPTION
**Pull request by jmacedoit**
This PR adds eslint rules to regulate method/properties in react classes.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -197,6 +197,49 @@ module.exports = {
     'react/prefer-es6-class': 'error',
     'react/react-in-jsx-scope': 'error',
     'react/self-closing-comp': 'error',
+    'react/sort-comp': [2 , {
```
Comment (bsonntag) [inline src/index.js]: Please use `'error'` instead of `2`.

File: src/index.js [hunk 2]
```diff
@@ -197,6 +197,49 @@ module.exports = {
...
+          'statics',
+          'defaultProps',
+          'constructor',
+          'getDefaultProps',
+          'state',
+          'getInitialState',
+          'getChildContext'
+        ],
+        lifecycle: [
+          'getDerivedStateFromProps',
```
Comment (bsonntag) [inline src/index.js]: This is a static method, so shouldn't it be in the `static-methods` group?

File: src/index.js [hunk 3]
```diff
@@ -197,6 +197,49 @@ module.exports = {
     'react/prefer-es6-class': 'error',
     'react/react-in-jsx-scope': 'error',
     'react/self-closing-comp': 'error',
+    'react/sort-comp': [2 , {
+      order: [
+        'static-methods',
+        'initialization',
+        'everything-else',
+        '/^on.+$/',
```
Comment (bsonntag) [inline src/index.js]: Event handle method names should start with `handle`, not with `on`, so should we have this on the rules?

**Comment by jmacedoit** [comment]
@bsonntag Please take another look.

**Pull request merged**